src/NServiceBus.Core/Pipeline/PipelineModelBuilder.cs

```diff
@@ -22,9 +23,23 @@ public List<RegisterStep> Build()
             var listOfBeforeAndAfterIds = new List<string>();
 
             // Let's do some validation too
+            if (addOrReplaceSteps.Any(addOrReplaceStep => addOrReplaceStep.RegisterStep.StepId != addOrReplaceStep.ReplaceStep.ReplaceId))
+            {
+                throw new Exception("Encountered RegisterOrReplace-registrations in the pipeline for which the ID differs between Add and Replace.");
```

Comment:
Add the ID's to the exception message?

Reply:
This is dropped as it can't occur anymore